src/Compilers/CSharp/Portable/Symbols/Synthesized/Extensions/SynthesizedExtensionMarker.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This method encodes the information needed to round-trip extension types
+    /// through metadata.
+    ///
+    /// It encodes:
+    /// - whether the extension type is implicit or explicit (PROTOTYPE)
+    /// - the underlying type (first parameter type)
+    /// - the base extensions (subsequent parameter types)
+    /// </summary>
+    internal sealed class SynthesizedExtensionMarker : SourceMemberMethodSymbol
+    {
+        private readonly TypeSymbol _returnType;
+        private readonly ImmutableArray<ParameterSymbol> _parameters;
+
+        internal SynthesizedExtensionMarker(SourceMemberContainerTypeSymbol containingType,
+            TypeSymbol underlyingType, ImmutableArray<NamedTypeSymbol> baseExtensionTypes, BindingDiagnosticBag diagnostics)
+            : base(containingType, syntaxReferenceOpt: containingType.SyntaxReferences[0], containingType.Locations[0], isIterator: false)
```

Comment:
```suggestion
            : base(containingType, syntaxReferenceOpt: containingType.SyntaxReferences[0], containingType.GetFirstLocation(), isIterator: false)
```

to be consistent with #67678